src/Evaluators/reduced_evaluator.jl

```diff
@@ -86,6 +86,43 @@ function ReducedSpaceEvaluator(
     )
 end
 
+function ReducedSpaceEvaluator(nlp::ReducedSpaceEvaluator, device = nothing) 
+    if device == CUDADevice()
```

Comment:
On a more generic standpoint (i.e. not necessarily related to this PR) I am wondering whether we should use our own types to handle the different devices, instead of KA's ones. Indeed, I am not sure how we could specify that we want to use both CPU and GPU in KA. An idea of implementation could be: 
```julia
abstract type AbstractExaDevice end

struct CPU <: AbstractExaDevice end

abstract type AbstractGPUDevice <: AbstractExaDevice end
struct ExaCUDA <: AbstractGPUDevice end
struct ExaROCm <: AbstractGPUDevice end
struct ExaOneAPI <: AbstractGPUDevice end

abstract type ExaMixedDevice <: AbstractExaDevice end
struct MixedGPUDevice <: ExaMixedDevice end

```
(just thinking out loud, I think we could improve this abstraction)

Reply:
Yes. Though we should maybe wait for KA to do the first step. But not sure...

Reply:
I will open a separate issue to discuss this